src/complex/Utilities/OStreamUtilities.cpp

```diff
@@ -0,0 +1,968 @@
+#include "OStreamUtilities.hpp"
+#include "StringUtilities.hpp"
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataObject.hpp"
+#include "complex/DataStructure/DataPath.hpp"
+#include "complex/DataStructure/DataStructure.hpp"
+#include "complex/DataStructure/IDataArray.hpp"
+#include "complex/DataStructure/INeighborList.hpp"
+#include "complex/DataStructure/NeighborList.hpp"
+#include "complex/DataStructure/StringArray.hpp"
+#include "complex/Utilities/ParallelDataAlgorithm.hpp"
+
+#include <algorithm>
+#include <fstream>
+#include <ostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+using namespace complex;
+
+void OStreamUtilities::OutputFunctions::neighborListImplWrapper(DataStructure& dataStructure, PrintMatrix2D& matrixRef, DataObject::Type type, const DataPath& path, bool hasIndex, bool hasHeaders)
+{
+  if(type == DataObject::Type::INeighborList) // unique because its loaded horizontally no dataAlg use
+  {
+    auto dataType = dataStructure.getDataAs<INeighborList>(path)->getDataType();
+    switch(dataType)
+    {
+    case DataType::int8: {
+      LoadNeighborListToMatrixImpl(matrixRef, dataStructure.getDataRefAs<NeighborList<int8>>(path), hasIndex, hasHeaders).loadToMatrix(0, dataStructure.getDataAs<NeighborList<int8>>(path)->getSize());
+      break;
+    }
```

Comment:
```suggestion
    const auto& iNeighborList = dataStructure.getDataRefAs<INeighborList>(path);
    auto dataType = iNeighborList.getDataType();
    switch(dataType)
    {
    case DataType::int8: {
      const auto& neighborList = dynamic_cast<const NeighborList<int8>&>(iNeighborList);
      LoadNeighborListToMatrixImpl(matrixRef, neighborList, hasIndex, hasHeaders).loadToMatrix(0, neighborList.getSize());
      break;
    }
```
This should store the neighbor list to avoid having to check the data structure multiple times. And since we know we have a `INeighborList` we can use `getDataRefAs`. This can be repeated for this rest of the cases.